docs/zkapps/tutorials/10-account-updates.mdx

```diff
@@ -0,0 +1,378 @@
+
+Last, to look briefly at [SecondaryZkApp.ts](https://github.com/o1-labs/docs2/tree/main/examples/zkapps/10-account-updates/src/SecondaryZkApp.ts), you have contents:
+
+```typescript
+export class SecondaryZkApp extends SmartContract {
+  @state(Field) num = State<Field>();
+
+  @method init() {
+    super.init();
+
+    this.account.provedState.assertEquals(this.account.provedState.get());
+    this.account.provedState.get().assertFalse();
+
+    this.num.set(Field(12));
+  }
+
+  @method add(incrementBy: Field) {
+    this.account.provedState.assertEquals(this.account.provedState.get());
+    this.account.provedState.get().assertTrue();
+
+    const num = this.num.get();
+    this.num.assertEquals(num);
+    this.num.set(num.add(incrementBy));
+  }
+}
+
+```
+
+You declare functions for initializing the account and the `add` method that is called from the earlier `ProofOnlyZkApp`.
+
+### Running our Smart Contracts and Visualizing the AccountUpdates
+
+Now it's time to learn about the [main.ts](https://github.com/o1-labs/docs2/tree/main/examples/zkapps/10-account-updates/src/main.ts) file that creates transactions with the earlier smart contracts and the account update visualizations it creates.
+
+First, import the transaction visualizer:
+
+```typescript
+...
+import { showTxn, saveTxn, printTxn } from 'mina-transaction-visualizer';
+...
+```
+
+This provides three functions:
+
+```typescript
+// creates a png file of a transaction, and opens it in a local image viewer
+async showTxn(txn: Mina.Transaction, name: string, legend: Legend)
+
+// creates a png file of a transaction, and saves it to a path
+saveTxn(txn: Mina.Transaction, name: string, legend: Legend, path: string)
+
+// prints a nicely formatted view of a transaction
+showTxn(txn: Mina.Transaction, name: string, legend: Legend)
+
+// with legend type, to replace public keys with human readable strings:
+type Legend = { [pk: string]: string };
+
+```
+
+Now build the legend as follows:
+
+```typescript
+  const legend = {
+    [proofsOnlyAddr.toBase58()]: 'proofsOnlyZkApp',
+    [secondaryAddr.toBase58()]: 'secondaryZkApp',
+    [deployerAccount.toPublicKey().toBase58()]: 'deployer',
+  };
+```
+
+Then deploy your smart contracts as follows and visualize the transaction:
+
+```typescript
+  const deployTxn = await Mina.transaction(deployerAccount, () => {
+    AccountUpdate.fundNewAccount(deployerAccount, 2);
+    proofsOnlyInstance.deploy();
+    secondaryInstance.deploy();
+  });
+
+  await deploy_txn.prove();
+  deploy_txn.sign([ deployerKey, proofsOnlySk, secondarySk ]);
+
+  await showTxn(deploy_txn, 'deploy_txn', legend);
+
+  await deploy_txn.send();
+```
+
+This yields the following visualization of `deploy_txn`. 
+
+
+<img src="/img/tutorial-10-deploy_txn.png" width="100%" />
+
+This is best opened in a new tab to read the contents ([link](/img/tutorial-10-deploy_txn.png)).
+
+The deploy transaction includes 5 accountUpdates. In order from left to right;
```

Comment:
```suggestion
The deploy transaction includes 5 accountUpdates represented as ovals. Described from left to right;
```

@es92 should this remain AccountUpdates (PascalCase) for consistency, or is the naming switching to how it's used in JavaScript?